xls/contrib/integrator/ir_integrator.h

```diff
@@ -21,6 +21,57 @@
 
 namespace xls {
 
+// Class that represents an integration function i.e. a function combining the
+// IR of other functions. This class tracks which orignal function nodes are
+// mapped to which integration function nodes. It also provides some utilities
+// that are useful for constructing the integrated function.
+class IntegrationFunction {
+ public:
+  IntegrationFunction() {}
+
+  // Create an IntegrationFunction object that is empty expect for
+  // paramters.
```

Comment:
sp: parameters

Reply:
Also maybe explain the "WithParamTuples" bit of the method name?

Reply:
Done.